benchmark/hetero/at/fusion.list.cpp.erb

```diff
@@ -2,24 +2,22 @@
 // Distributed under the Boost Software License, Version 1.0.
 // (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
 
-<% if n > 10 %>
-    #define FUSION_MAX_LIST_SIZE <%= ((n + 9) / 10) * 10 %>
-<% end %>
-
 #include <boost/fusion/include/at.hpp>
 #include <boost/fusion/include/make_list.hpp>
 namespace fusion = boost::fusion;
 
-
-template <int>
-struct x { };
+template<int> struct x {};
 
 int main() {
-    auto xs = fusion::make_list(
-        <%= (1..n).map { |n| "x<#{n}>()" }.join(', ') %>
+<% env[:k].times do |k| %>
+    /*constexpr*/ auto <%= "xs#{k}" %> = fusion::make_list(
+        <%= ((k*n+1)..(k*n+n)).map { |n| "x<#{n}>()" }.join(', ') %>
     );
 
-#if defined(METABENCH)
-    fusion::at_c<<%= n - 1 %>>(xs);
-#endif
+    <% (env[:p]*n/100).times do |p| %>
+    #if defined(METABENCH)
+        fusion::at_c<<%= p*100/env[:p] %>>(<%= "xs#{k}" %>);
```

Comment:
Not exactly, that actually means `p%` elements equally distributed in the range `[0, n-1]` are retrieved, so, for instance, if `n = 60` and `p = 10%`, then indices will be `[0, 10, 20, 30, 40, 50]`.